src/views/Home.jsx

```diff
@@ -17,15 +20,55 @@ export function Home({ createToken, setListToken }) {
 		} else {
 			listId = null;
 			setListToken(listId);
-			setMessage('Your shopping list was not created. Please try again. ');
+			setCreateListMessage(
+				'Your shopping list was not created. Please try again. ',
+			);
 		}
 	}
 
+	async function handleJoinClick(e) {
+		e.preventDefault();
+
+		if (tokenInput) {
+			let listId = tokenInput;
+			const listExists = await checkIfListExists(listId);
+
+			if (listExists) {
```

Comment:
**Suggestion:** Refactor to remove nested If statements.

I've heard that writing nested If statements like this is a bad smell, because it adds unnecessary complexity. I propose removing this 2nd `If` (lines 36-41) from living within the 1st one.

Reply:
We refactored this a bit. Take a look and let us know what you think.